### PR TITLE
Add test for negative instanceof variable outside if scope

### PR DESCRIPTION
**Pull request by vegegoku**
This pull request address the comment here 
https://github.com/gwtproject/gwt/pull/9917#issuecomment-2274516384

### CODE REVIEW

File: user/test-super/com/google/gwt/dev/jjs/super/com/google/gwt/dev/jjs/test/Java17Test.java [hunk 1]
```diff
@@ -212,4 +212,12 @@ public void testInstanceOfPatternMatchingAsReturn() {
     Bar bar = new Bar();
     assertTrue(bar.isSquare());
   }
+
+  public void testNegativeInstanceOfPatternOutsideIfScope() {
+    Object bar = new Bar();
+    if(!(bar instanceof Bar b)) {
```
Comment (zbynek) [inline user/test-super/com/google/gwt/dev/jjs/super/com/google/gwt/dev/jjs/test/Java17Test.java:218]:
```suggestion
    if (!(bar instanceof Bar b)) {
```

**Review by niloc132 — APPROVED** [review]
